### DISCUSSION TITLE
Bun: support for private registries -> bunfig.toml

### DISCUSSION THREAD

**Discussion by DanielRudrich** [vercel/community#4091] [Idea]
### Question

Hi all,

first and foremost: great to see that there's [official support](https://vercel.com/changelog/bun-install-is-now-supported-with-zero-configuration) for [Bun](https://bun.sh) 🎉 

Is there / will there be support for private package registries when using bun on vercel?

For NPM this seems to be possible using the [`NPM_RC` environment variable](https://vercel.com/guides/using-private-dependencies-with-vercel#other-package-registries) to access private package registries. Unfortunately, bun [doesn't use the .npmrc config](https://bun.sh/docs/install/registries#npmrc) (yet?) but has its own configuration file [`bunfig.toml`](https://bun.sh/docs/install/registries). 

Are there plans for a `BUNFIG` environment variable or is this already available?

Cheers
Daniel

**Comment by kierancrown** [comment]
I'd also be interested in this as we can't ship with bun until this is supported 🙏

**Comment by amyegan** [comment]
We don't currently have support for a `BUNFIG` environment variable, but this is a nice idea! I'm sharing it with the team 🙇

**Reply by ChickenBone** [reply]
Yes please! Just ran into this today trying to switch over to bun! Last roadblock for me!

**Comment by filipditrich** [comment]
Hi to all,
had the same problem but managed to make it working.

You can do so by leveraging the "Install command" at Vercel's Project Settings:
<img width="940" alt="Screenshot 2024-01-21 at 11 10 16 AM" src="https://github.com/vercel/community/assets/35503864/f61157fa-66c6-42f4-b8be-b86dee18f5e9">

There in the script you have access to all the env variables defined in your project, so you can easily just update/create the `bunfig.toml` file with, like below:
```bash
#!/bin/bash

# add bunfig.toml file with $BUNFIG env variable
echo "⏳ Adding bunfig.toml file..."
echo "$BUNFIG" > bunfig.toml
echo "✅ Done adding bunfig.toml file."

# run install
echo "⏳ Running install..."
bun install
echo "✅ Done running install."
```

Hope it helps!

**Reply by tlhfckoctbcr** [reply]
Thank you for the tip. Took your advice and built this with Bun Shell:

```
import { $ } from 'bun';

const bunfigContent = `[install.scopes]
\"@privatelib\" = { token = \"${process.env.NPM_TOKEN}\", url = \"https://registry.npmjs.com/\" }`;

await $`echo "Creating bunfig.toml..."`;
await $`rm -f bunfig.toml`;
await $`echo ${bunfigContent} > bunfig.toml`;
await $`echo "Bunfig created. Running installation..."`;
await $`bun i`;
await $`echo "Installation complete."`;
```
